config/crds/v1beta1/patches/agent-patches.yaml

```diff
@@ -25,3 +25,6 @@
   path: /spec/validation/openAPIV3Schema/properties/spec/properties/deployment/properties/strategy/properties/rollingUpdate/properties/maxUnavailable/x-kubernetes-int-or-string
 - op: remove
   path: /spec/validation/openAPIV3Schema/properties/spec/properties/daemonSet/properties/updateStrategy/properties/rollingUpdate/properties/maxUnavailable/x-kubernetes-int-or-string
+# we need to generate x-kubernetes-preserve-unknown-fields for v1 CRDs but they break v1beta so we have to remove them again here
+- op: remove
+  path: /spec/validation/openAPIV3Schema/properties/spec/properties/config/x-kubernetes-preserve-unknown-fields
```

Comment:
Normal that we don't have the same count of `x-kubernetes-preserve-unknown-fields` and `// +kubebuilder:pruning:PreserveUnknownFields`?

Agent and Beat do not get:
```
- op: remove
  path: /spec/validation/openAPIV3Schema/properties/spec/properties/daemonSet/properties/podTemplate/x-kubernetes-preserve-unknown-fields
- op: remove
  path: /spec/validation/openAPIV3Schema/properties/spec/properties/deployment/properties/podTemplate/x-kubernetes-preserve-unknown-fields
```

Reply:
That's because we remove the whole `podTemplate` section for the `v1beta1` version on the top of the file.

Reply:
Of course.
That's not important but just for my understanding, we could do the same the following patches where we also remove the whole `podTemplate`?
```
v1beta1/patches/apm-kibana-patches.yaml:  path: /spec/validation/openAPIV3Schema/properties/spec/properties/podTemplate/properties
v1beta1/patches/apm-kibana-patches.yaml:  path: /spec/validation/openAPIV3Schema/properties/spec/properties/podTemplate/x-kubernetes-preserve-unknown-fields

v1beta1/patches/elasticsearch-patches.yaml:  path: /spec/validation/openAPIV3Schema/properties/spec/properties/nodeSets/items/properties/podTemplate/properties
v1beta1/patches/elasticsearch-patches.yaml:  path: /spec/validation/openAPIV3Schema/properties/spec/properties/nodeSets/items/properties/podTemplate/x-kubernetes-preserve-unknown-fields

v1beta1/patches/ems-patches.yaml:  path: /spec/validation/openAPIV3Schema/properties/spec/properties/podTemplate/properties
v1beta1/patches/ems-patches.yaml:  path: /spec/validation/openAPIV3Schema/properties/spec/properties/podTemplate/x-kubernetes-preserve-unknown-fields

v1beta1/patches/ent-patches.yaml:  path: /spec/validation/openAPIV3Schema/properties/spec/properties/podTemplate/properties
v1beta1/patches/ent-patches.yaml:  path: /spec/validation/openAPIV3Schema/properties/spec/properties/podTemplate/x-kubernetes-preserve-unknown-fields
```

Reply:
The way we remove the `podTemplate` differs a little bit between Agent and Beats and the other CRDs. Agent and Beats remove the whole `podTemplate` tree, while the other CRDs only remove the `properties` from the `podTemplate` section. This makes the other tweaking necessary. This is nothing this PR introduced but has been in place since the introduction of Beats and Agent. I kept it that way so that the legacy CRDs stay as close to the versions we currently ship as possible.